### DISCUSSION TITLE
Embedded languages in tagged templates

### DISCUSSION THREAD

**Discussion by aral** [phcode-dev/phoenix#762] [Ideas]
Just testing out Phoenix, possibly as a good tool for education when working on [Kitten](https://codeberg.org/kitten/app) projects in the future, and it would be lovely to have syntax highlighting and language intelligence for embedded languages within tagged templates (as they’re used everywhere in it).

Specifically, for HTML and CSS:

```js
const markup = html`
  <h1>I’m HTML</h1>
`
const styles = css`
   h1 { font-size: large; }
`
```

Haven’t looked into the code so I don’t know how easy/possible it is to implement but it seems like the typescript-language-server LSP is able to handle them at least (I’m using it in [Helix Editor](https://helix-editor.com) currently).

**Comment by abose** [comment]
Created https://github.com/phcode-dev/phoenix/issues/767 to track.
Phoenix doesn't use LSP yet and relies on our own language intelligence modules. But the current plan is to add LSP as soon as work related to native windows/mac builds is complete.

**Comment by aral** [comment]
@abose Thanks. Also, I was imprecise: it should be the tree sitter grammars in Helix Editor that provide the syntax highlighting not the LSP.
